File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/auctions/dto/AuctionResponse.java
package com.portfolio.auctionmarket.domain.auctions.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.portfolio.auctionmarket.domain.auctions.entity.Auction;
import com.portfolio.auctionmarket.domain.auctions.entity.AuctionStatus;
import com.portfolio.auctionmarket.domain.products.entity.Product;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuctionResponse {

    private Long auctionId;

    private Long startPrice;

    private Long currentPrice;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime startTime;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime endTime;

    private AuctionStatus status;

    public static AuctionResponse from(Auction entity) {
        return AuctionResponse.builder()
                .auctionId(entity.getAuctionId())
                .startPrice(entity.getStartPrice())
                .currentPrice(entity.getCurrentPrice())
                .startTime(entity.getStartTime())
                .endTime(entity.getEndTime())
                .status(entity.getStatus())
                .build();
    }
}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/auction/dto/AuctionResponse.kt
package org.example.auctioncommon.domain.auction.dto

import com.fasterxml.jackson.annotation.JsonFormat
import org.example.auctioncommon.domain.auction.entity.Auction
import org.example.auctioncommon.domain.auction.entity.AuctionStatus
import java.time.LocalDateTime

data class AuctionResponse(
    val auctionId: Long,
    val startPrice: Long,
    val currentPrice: Long,
    val status: AuctionStatus,
    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val startTime: LocalDateTime,
    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val endTime: LocalDateTime
)

fun Auction.toAuctionResponse(): AuctionResponse {
    return AuctionResponse(
        auctionId = this.auctionId ?: 0L,
        startPrice = this.startPrice ?: 0L,
        currentPrice = this.currentPrice ?: 0L,
        startTime = this.startTime ?: LocalDateTime.now(),
        endTime = this.endTime ?: LocalDateTime.now(),
        status = this.status ?: AuctionStatus.PROCEEDING,
    )
}